src/redux/Categories/categoriesSlice.js

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: [],
+};
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state, action) => {
+      const newState = {
+        ...state,
+        categories: action.payload === 'Under construction'
+          ? 'Under construction'
+          : state.categories,
+      };
+
+      // return the new state object
+      return newState;
+    },
+  },
+});
```

Comment:
- [OPTIONAL] Great job so far. I would recommend you to look at below code snippet. It looks more cleaner.

```
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```